### PR TITLE
Updates Section 6.1.2 - "Registration Objects" for Applications.

### PR DESCRIPTION
**Pull request by cpu**
In [Section 6.1.2 "Registration Objects"](https://github.com/ietf-wg-acme/acme/blob/83e2f7985f5921a4ac30b7446dbb9c1d347d69ec/draft-ietf-acme-acme.md#registration-objects) the "applications" field of the registration resource is described in great detail. This is confusing for two reasons:

1. both the text and the example registration object are still using the "authorization" term that applications replace.
2. the text is written as if the applications should be directly embedded in the registration object returned.

This PR addresses both issues by describing the "applications" field as an array of application URIs that the client can fetch. The example "registration" object is updated accordingly.

The note indicating a server SHOULD include pending applications and SHOULD NOT include invalid applications remains as-is. The note indicating a server MAY return an incomplete list is removed - it is more complicated and with only URIs being returned, likely an unneeded optimization.

Finally, in [Section 6.3 "Registration"](https://github.com/ietf-wg-acme/acme/blob/83e2f7985f5921a4ac30b7446dbb9c1d347d69ec/draft-ietf-acme-acme.md#registration) there are three fields that must be ignored by the server in a registration body: key, authorizations and certificates. This should be "key" and "applications" and is updated accordingly.

### CODE REVIEW

File: draft-ietf-acme-acme.md [hunk 1]
```diff
@@ -768,6 +762,8 @@ syntax like "confirm": true ]]
 support reissuance / renewal in a straightforward way, especially if the CSR /
 notBefore / notAfter could be updated. ]]
 
+#### Application Requirements
+
```
Comment (bifurcation) [inline draft-ietf-acme-acme.md]: Note: These changes are unrelated to the topic of the PR.  I think they're OK, though.

File: draft-ietf-acme-acme.md [hunk 2]
```diff
@@ -818,6 +814,30 @@ url (required, string):
 To fulfill this requirement, the ACME client should direct the user to the
 indicated web page.
 
+#### Applications List
```
Comment (bifurcation) [inline draft-ietf-acme-acme.md]: I agree that this is better as its own subsection, but it would go better under the "Registration Objects" section, since ultimately it's about how registrations refer to applications.

File: draft-ietf-acme-acme.md [hunk 3]
```diff
@@ -818,6 +814,30 @@ url (required, string):
...
+~~~~~~~~~~
+HTTP/1.1 200 OK
+Content-Type: application/json
+Link: href="/acme/reg/1/apps?cursor=2", rel="next"
+
+{
+  "applications": [
+    "https://example.com/acme/reg/1/apps/1",
+    "https://example.com/acme/reg/1/apps/2"
+  ]
```
Comment (bifurcation) [inline draft-ietf-acme-acme.md]: For a little more verisimilitude, you might up the numbers a little bit :)

```
HTTP/1.1 200 OK
Content-Type: application/json
Link: href="/acme/reg/1/apps?cursor=51", rel="next"

{
  "applications": [
    "https://example.com/acme/reg/1/apps/1",
    "https://example.com/acme/reg/1/apps/2",
    /* 47 more URLs */
    "https://example.com/acme/reg/1/apps/50"
  ]
}
```

Comment (cpu) [inline draft-ietf-acme-acme.md]: Good idea ("verisimilitude" is an excellent word!!!)

**Comment by cpu** [comment]
@bifurcation feedback addressed, and conflicts w/ master resolved. Thanks for the :eyes: !
